src/Pulsar.Client/Internal/ClientCnx.fs

```diff
@@ -212,73 +209,75 @@ type ClientCnx (broker: Broker,
-                        sendMb.Post(SocketMessageWithoutReply (Commands.newPong()))
-                        reader.AdvanceTo(consumed)
-                    | XCommandMessage ((cmd, metadata, payload), consumed) ->
-                        let consumerMb = consumers.[%cmd.ConsumerId]
-                        consumerMb.Post(MessageRecieved { MessageId = MessageId.FromMessageIdData(cmd.MessageId); Payload = payload })
-                        reader.AdvanceTo(consumed)
-                    | XCommandLookupTopicResponse (cmd, consumed) ->
-                        let result =
-                            if (cmd.ShouldSerializeError())
-                            then
-                                Log.Logger.LogError("Error: {0}. Message: {1}", cmd.Error, cmd.Message)
-                                Error
-                            else
-                                LookupTopicResult { BrokerServiceUrl = cmd.brokerServiceUrl; Proxy = cmd.ProxyThroughServiceUrl }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandProducerSuccess (cmd, consumed) ->
-                        let result = ProducerSuccess { GeneratedProducerName = cmd.ProducerName }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandSuccess (cmd, consumed) ->
-                        handleRespone %cmd.RequestId Empty reader consumed
-                    | XCommandCloseProducer (cmd, consumed) ->
-                        let producerMb = producers.[%cmd.ProducerId]
-                        producers.Remove(%cmd.ProducerId) |> ignore
-                        producerMb.Post(ProducerMessage.ConnectionClosed)
-                        reader.AdvanceTo(consumed)
-                    | XCommandCloseConsumer (cmd, consumed) ->
-                        let consumerMb = consumers.[%cmd.ConsumerId]
-                        consumers.Remove(%cmd.ConsumerId) |> ignore
-                        consumerMb.Post(ConsumerMessage.ConnectionClosed)
-                        reader.AdvanceTo(consumed)
-                    | XCommandReachedEndOfTopic (cmd, consumed) ->
-                        let consumerMb = consumers.[%cmd.ConsumerId]
-                        consumerMb.Post ReachedEndOfTheTopic
-                        reader.AdvanceTo(consumed)
-                    | XCommandGetTopicsOfNamespaceResponse (cmd, consumed) ->
-                        let result = TopicsOfNamespace { Topics = List.ofSeq cmd.Topics }
-                        handleRespone %cmd.RequestId result reader consumed
-                    | XCommandError (cmd, consumed) ->
-                        Log.Logger.LogError("Error: {0}. Message: {1}", cmd.Error, cmd.Message)
-                        let result = Error
-                        handleRespone %cmd.RequestId result reader consumed
-                    | IncompleteCommand ->
+                    | Result.Error IncompleteCommand ->
```

Comment:
No need to handle errors before Ok results, they happen much less often

Reply:
see my first comment